src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java

```diff
@@ -11,56 +11,49 @@
 import org.apache.commons.compress.utils.IOUtils;
 import org.springframework.beans.factory.InitializingBean;
 
-import com.arondor.arender.viewer.common.rest.DocumentServiceRestClient;
-import com.arondor.viewer.client.api.document.DocumentId;
-import com.arondor.viewer.common.document.id.DocumentIdFactory;
-import com.arondor.viewer.rendition.api.document.DocumentAccessorSelector;
-
 public class ARenderContentTransformerWorker implements ContentTransformerWorker, InitializingBean
 {
-    private DocumentServiceRestClient restClient = new DocumentServiceRestClient();
+    private final ContentTransformARenderRestClient restClient = new ContentTransformARenderRestClient();
+
+    private String arenderRenditionServerAddress = null;
 
-    private String arenderRenditionServerAddress = "http://localhost:8761/";
+    public ARenderContentTransformerWorker() {
+        restClient.setAddress(arenderRenditionServerAddress);
```

Comment:
Do not call the setter here, since the variable has not yet been set at the time the constructor is called.

Reply:
@vierbergenlars when does the variable arenderRenditionServerAddress get the value from alfresco-global.properties??  (i was thinking the first thing done is setting all variables from alfresco-global.properties and then running this code isn't?

Reply:
Well, the "properties" defined in the xml are set by calling the setter.
Before you can call a setter on an object, it first needs to exist, so the constructor has to be called to construct the object.

After those properties have been set, then `afterPropertiesSet()` is called by spring, and your properties have the correct value.